replica/database.cc

```diff
@@ -531,6 +531,20 @@ void backlog_controller::update_controller(float shares) {
     _scheduling_group.set_shares(shares);
 }
 
+void compaction_controller::update_controller(float shares) {
+    _computed_shares = shares;
+    if (_max_shares > 0.0f && shares > _max_shares) {
+        shares = _max_shares;
+    }
+    backlog_controller::update_controller(shares);
+}
+
+void compaction_controller::update_max_shares(float max_shares) {
+    _max_shares = max_shares;
+    // Update the shares of the controller w.r.to the new max shares.
+    update_controller(_computed_shares);
+}
```

Comment:
we don't really need to do this. see backlog_controller::update_static_shares(). once we update a setting which changes controller update behavior, then the next calculation cycle will take care of it, happens in sub second interval.